internal/resultstore/resultstore.go

```diff
@@ -56,6 +61,40 @@ func (rs *ResultStore) generatePath(p Pkg) string {
 	return path
 }
 
+func (rs *ResultStore) SaveWriteBufferZip(ctx context.Context, p Pkg, fileName string, writeBufferZip *os.File) error {
+	path := filepath.Join(rs.generatePath(p), writeBufferFolder)
+	bkt, err := blob.OpenBucket(ctx, rs.bucket)
+	if err != nil {
+		return err
+	}
+	defer bkt.Close()
+
+	uploadPath := filepath.Join(path, fileName+".zip")
+	log.Info("Uploading results",
+		"bucket", rs.bucket,
+		"path", uploadPath)
+
+	w, err := bkt.NewWriter(ctx, uploadPath, nil)
+	if err != nil {
+		return err
+	}
+	zipFile, err := os.OpenFile(writeBufferZip.Name(), os.O_RDWR, 0666)
```

Comment:
We can potentially bypass having a zipfile on disk completely, if we just write the zip using https://pkg.go.dev/archive/zip#Writer right?

We can wrap the `io.Writer` from the `bkt.NewWriter` call, and just directly write the zip contents to GCS.